design/healcheck-endpoint.md

```diff
@@ -0,0 +1,47 @@
+# Healthcheck endpoint
+
+* **Status**: Draft #1
+* **JIRA**: [KEYCLOAK-12398](https://issues.redhat.com/browse/KEYCLOAK-12398)
+
+## Motivation
+
+Deploying both Keycloak and RHSSO on top of Kubernetes requires defining both Readiness Check and Liveness Check (see the [manual](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/)). Currently for Keycloak we use a very naive implementation, that calls `/auth/realms/master` URL. This approach has many drawbacks, here are a few examples:
+
+* We can not discover database failures
+* We can not discover if certain Identity Provider is up and running
+* Undertow subsystem is down when proceeding with migration
+
+In order to address this problems, we should provide a parametrized Healthcheck endpoint in the Server and use it in our container image as well as in the Operator. 
+
+## Parametrizing the Healthcheck URL
```

Comment:
I think there's also a case for check, but not fail on. So something like:

`include-checks=db,idp-google&fail-on=db`

Reply:
I think this gap has just been filled by the https://github.com/keycloak/keycloak-community/pull/177

Reply:
Addressed in the next iteration. Resolving.